lib/ffmpeg/movie.rb

```diff
@@ -217,14 +217,29 @@ def fix_encoding(output)
       output.force_encoding("ISO-8859-1")
     end
 
-    def head
-      url = URI(@path)
+    def head(location=@path, limit=FFMPEG.max_http_redirect_attempts)
+      url = URI(location)
       return unless url.path
 
       http = Net::HTTP.new(url.host, url.port)
       http.use_ssl = url.port == 443
-      http.request_head(url.path)
-    rescue SocketError, Errno::ECONNREFUSED
+      response = http.request_head(url.path)
+
+      case response
+        when Net::HTTPRedirection then
+          raise FFMPEG::HTTPTooManyRequests if limit == 0
```

Comment:
I'm opting to validate the setting, following the pattern already set: >= 0 and a whole number.